x-pack/plugins/entity_manager/server/plugin.ts

```diff
@@ -185,9 +187,19 @@ export class EntityManagerServerPlugin
       .catch((err) => this.logger.error(err));
 
     // Setup v2 definitions index
-    setupEntityDefinitionsIndex(core.elasticsearch.client, this.logger).catch((error) => {
-      this.logger.error(error);
-    });
+    setupEntityDefinitionsIndex(core.elasticsearch.client, this.logger)
+      .then(() => {
+        // Using asScope with a FakeRequest to get a IScopedClusterClient which the lib functions expect
+        // They internally grab the internal user to use
+        const fakeRequest: FakeRequest = { headers: {} };
+        return installBuiltInDefinitions(
+          core.elasticsearch.client.asScoped(fakeRequest),
```

Comment:
As I commented above:
```
// Using asScope with a FakeRequest to get a IScopedClusterClient which the lib functions expect
// They internally grab the internal user to use
 ```

So I cannot grab the one on line 170 because it's `IClusterClient`

Reply:
do we need ClusterClient to install built in definitions ? iiuc we can only do that with internal user because of the dot-index, so a prescoped ElasticsearchClient should be enough ?

Reply:
It's because in the routes we use the EntityClient which is scoped so it uses `IScopedClusterClient` so that is what it passes to the lib functions, I can try to change the types and see if it still works

Reply:
No, they don't overlap so I cannot cast it :/

Reply:
Maybe I'm being overly cautious but I prefer if we pass around (versions of) Cluster client so that the choice between internal user and scoped user can be done more clearly and visibly in the code, which will also make it easier to audit. Once we get `ElasticsearchClient` we can no longer tell which credentials are being used :/

Reply:
I can add an type like `Pick<IClusterClient, 'asInternalUser'>` which would allow us to pass around any of the cluster client but still audit where we grab the internal user, WDYT?

Reply:
what happens when calling `.asCurrentUser` on a cluster client scoped with a fake request ? does it throw or falls back to internal user ?

> so that the choice between internal user and scoped user can be done more clearly and visibly in the code

as in closer to the calling code ? because that decision still has to be made somewhere, but probably farther from the call site